PowerEditor/src/NppIO.cpp

```diff
@@ -1969,6 +1969,86 @@ bool Notepad_plus::fileRename(BufferID id)
 	return success;
 }
 
+bool Notepad_plus::fileRenameUntitled(BufferID id /* =BUFFER_INVALID */, const wchar_t* tabNewName /* =nullptr */)
+{
+	BufferID bufferID = id;
+	if (id == BUFFER_INVALID)
+	{
+		bufferID = _pEditView->getCurrentBufferID();
+	}
+	Buffer* buf = MainFileManager.getBufferByID(bufferID);
+
+	SCNotification scnN{};
+	scnN.nmhdr.code = NPPN_FILEBEFORERENAME;
+	scnN.nmhdr.hwndFrom = _pPublicInterface->getHSelf();
+	scnN.nmhdr.idFrom = (uptr_t)bufferID;
+	_pluginsManager.notify(&scnN);
+
+	bool success = false;
+
+	bool isFileExisting = PathFileExists(buf->getFullPathName()) != FALSE;
+	if (!isFileExisting)
+	{
+		// We are just going to rename the tab nothing else
+		// So just rename the tab and rename the backup file too if applicable
+
+		if (tabNewName != nullptr)
+		{
+			std::wstring tabNewNameStr = tabNewName;
+
+			trim(tabNewNameStr); // No leading and tailing space allowed
+
+			if (!tabNewNameStr.empty())
+			{
+				if (tabNewNameStr.length() < langNameLenMax)
+				{
+					// https://docs.microsoft.com/en-us/windows/desktop/FileIO/naming-a-file
+					// Reserved characters:  < > : " / \ | ? * tab  
+					//  ("tab" is not in the official list, but it is good to avoid it)
+					const std::wstring reservedChars = TEXT("<>:\"/\\|\?*\t");

```

Comment:
Use the same RegExpr here:
https://github.com/notepad-plus-plus/notepad-plus-plus/blob/master/PowerEditor/src/NppIO.cpp#L1906

Please define a symbol and use it in the both places.

Reply:
I wasn't sure where to put it, so I put it close to top-of-file -- hope that is reasonable.

I also defined a shared constant for the maximum length of the name.  Existing code has it using the max length for a language name, but this seems unrelated, so I gave it its own constant.